modules/core/src/main/java/org/apache/ignite/internal/commandline/Command.java

```diff
@@ -56,12 +61,78 @@ public static GridClient startClient(GridClientConfiguration clientCfg) throws E
      *
      * @param logger Logger to use.
      * @param desc Command description.
+     * @param cmd Command.
      * @param args Arguments.
      */
     public static void usage(Logger logger, String desc, CommandList cmd, String... args) {
+        usage(logger, desc, cmd, null, args);
+    }
+
+    /**
+     * Print command usage.
+     *
+     * @param logger Logger to use.
+     * @param desc Command description.
+     * @param cmd Command.
+     * @param paramsDesc Description of parameters (optional).
+     * @param args Arguments.
+     */
+    public static void usage(
+        Logger logger,
+        String desc,
+        CommandList cmd,
+        @Nullable Map<String, String> paramsDesc,
+        String... args
+    ) {
+        logger.info("");
         logger.info(INDENT + desc);
         logger.info(DOUBLE_INDENT + CommandLogger.join(" ", UTILITY_NAME, cmd, CommandLogger.join(" ", args)));
-        logger.info("");
+
+        if (!F.isEmpty(paramsDesc)) {
+            logger.info("");
+
+            logger.info(DOUBLE_INDENT + "Parameters:");
+
+            usageParams(paramsDesc, DOUBLE_INDENT + INDENT, logger);
+        }
+    }
+
+    /**
+     * Print cache command arguments usage.
+     *
+     * @param paramsDesc Cache command arguments description.
+     * @param indent Indent string.
+     * @param logger Logger to use.
+     */
+    public static void usageParams(Map<String, String> paramsDesc, String indent, Logger logger) {
+        int maxParamLen = paramsDesc.keySet().stream().max(Comparator.comparingInt(String::length)).get().length();
+
+        for (Map.Entry<String, String> param : paramsDesc.entrySet())
+            logger.info(indent + extendToLen(param.getKey(), maxParamLen) + "  " + "- " + param.getValue());
+    }
+
+    /**
+     * Appends spaces to end of input string for extending to needed length.
+     *
+     * @param s Input string.
+     * @param targetLen Needed length.
+     * @return String with appended spaces on the end.
+     */
+    public static String extendToLen(String s, int targetLen) {
+        assert targetLen >= 0;
+        assert s.length() <= targetLen;
+
+        if (s.length() == targetLen)
+            return s;
+
+        SB sb = new SB(targetLen);
+
+        sb.a(s);
+
+        for (int i = 0; i < targetLen - s.length(); i++)
+            sb.a(" ");
```

Comment:
String.join(s, Collections.nCopies(n, " ")); may be so ?

Reply:
@zstan Your suggestion requires more memory allocation.

Reply:
I'd prefer to keep it as is.